src/tls/boring.rs

```diff
@@ -162,6 +164,50 @@ impl Certs {
     }
 }
 
+#[derive(Clone)]
+enum Verifier {
+    // Does not verify the identity.
+    None,
+
+    // Allows exactly one identity
+    Exact(Identity),
+    // TODO other types could be allowlist/denylist or some combo for RBAC
```

Comment:
I suspect RBAC will probably be denied at the HBONE request level. But just my initial impression